tracer/src/Datadog.Trace/Telemetry/Collectors/DependencyTelemetryCollector.cs

```diff
@@ -70,13 +61,14 @@ public bool HasChanges()
         /// <returns>Null if there are no changes, or the collector is not yet initialized</returns>
         public ICollection<DependencyTelemetryData> GetData()
         {
-            var hasChanges = Interlocked.CompareExchange(ref _hasChangesFlag, 0, 1) == 1;
-            if (!hasChanges)
+            var assemblies = Interlocked.Exchange(ref _assemblies, new ConcurrentDictionary<DependencyTelemetryData, bool>());
```

Comment:
This change in behaviour will break all our assumptions in the telemetry checks in the integration tests, so those will all need refactoring

Reply:
As discussed, it's not the case for dependencies